24H Study Room/ViewModels/LoginViewModel.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  LoginViewModel.swift
+//  24H Study Room
+//
+//  Created by Jia-Jiuan Tsai on 2022-09-05.
+//
+
+import Foundation
+
+class LoginViewModel: ObservableObject {
+    private(set) var fireAuthService: FireAuthServicing
+    @Published var inputEmail = ""
+    @Published var inputPassword = ""
+    @Published var emailIsValid = false
+    @Published var email: Email?
```

Comment:
[naming] Perhaps it would be clearer if these were named `validEmail` and `validPassword`, so that they're not confused with the other unvalidated fields